README.md

```diff
@@ -1,15 +1,314 @@
+
+The following is almost a copy & paste of the
+[original ethers documentation](https://docs.ethers.org/v6/getting-started/#starting-connecting).
+Pay attention to the creation of the `ShutterProvider` - you need to pass the `options` created
+earlier.
+
+```js
+let signer = null
+
+let provider
+if (window.ethereum == null) {
+  // If MetaMask is not installed, we use the default provider,
+  // which is backed by a variety of third-party services (such
+  // as INFURA). They do not have private keys installed,
+  // so they only have read-only access
+  console.log('MetaMask not installed; using read-only defaults')
+  provider = ethers.getDefaultProvider()
+} else {
+  // Connect to the MetaMask EIP-1193 object. This is a standard
+  // protocol that allows Ethers access to make all read-only
+  // requests through MetaMask.
+  // SHUTTER> use ShutterProvider
+  provider = new ShutterProvider(options, window.ethereum)
+
+  // It also provides an opportunity to request access to write
+  // operations, which will be performed by the private key
+  // that MetaMask manages for the user.
+  signer = await provider.getSigner()
+}
+```
+
+#### Encrypt a transaction
+
+Sending an encrypted transaction can now be achieved through the standard interface,
+`signer.sendTransaction()`. This will trigger the usual signature request in the wallet plugin.
+
+```js
+// transaction parameters
+let txRequest = {
+  from: signer.address,
+  to: signer.address,
+  value: 1,
+  data: '0x',
+}
+
+// send transaction with standard settings
+const txResponse = await signer.sendTransaction(txRequest)
+
+// wait for the inboxTransaction to be mined:
+await txResponse.wait()
+```
+
+**IMPORTANT**: The transaction presented in the confirmation dialog will be different from the
+content specified in `txRequest`. This is due to the request being encrypted and sent to the
+`Inbox` contract. You will need to communicate to your users what is happening!
+
+ShutterSigner will automatically
+
+- query the encryption parameters
+- schedule an execution block based on the `inclusionWindow` parameter (default: `25`)
+  `scheduledExecutionBlock = latestBlock + inclusionWindow`
+- encrypt the original transaction
+- estimate the necessary gas
+- prepare a transaction to the `Inbox` contract
+- trigger the user interaction for signing
+
+**IMPORTANT**: if the user does not sign the transaction in time to be included
+before `scheduledExecutionBlock`, the transaction **will** fail.
+
+#### Tracing the encrypted transaction submission
+
+In order to present more status to the user (i.e. time to execution, encrypted message, ...),
+there is also a more comprehensive function to send transactions, `signer._sendTransactionTrace()`:
+
+```js
+const inclusionWindow = 25 // default inclusionWindow
+const [encryptedMessage, txResponse, scheduledExecutionBlock] =
+  await signer._sendTransactionTrace(txRequest, inclusionWindow)
+```
+
+#### Inspecting shutterized blocks
```

Comment:
Is this explanation valid and usable if there were multiple encrypted transactions in a block or is this specific to a single encrypted transaction for this epoch?

Reply:
yes, this should trace all `executions` (note the plural) of encrypted transactions

Reply:
Ok what would it take to filter logs for a specific transaction? I think that's what people want to have